### ISSUE TITLE
`KeyError: 'Replacement'` thrown by interactive.py when stacker build used with `--replacements-only` on simple SG addition

### ISSUE THREAD

**Issue by GarisonLotus** [remind101/stacker#463]
tested with Stacker 1.0.4

Our automated builds get triggered based on changes in a git repo. To be safe, we execute the stacker build with the `--replacements-only` flag in interactive mode. I ran into an issue and I have narrowed down a repeatable test cause where I could cause stacker to crash when I simply tried to add a single additional security group rule to an existing ruleset in a CFN stack. All changed was a new single security group rule added in a quick loop I created.    Below is the output of my stacker run in jenkins:

```
15:50:30 --------------------------------------------------------------------------------
15:50:30 building the following environment...
15:50:30 --------------------------------------------------------------------------------
15:50:30 The environment is: cgm
15:50:30 The stacker-config is: z_garison_test-core.yaml
15:50:30 The region is: us-east-1
15:50:30 
15:50:30 tags:
15:50:30   "team": Operations
15:50:30   "project": garison-test
15:50:30   "component": garison-test
15:50:30   "stacker_namespace": ${namespace}
15:50:30 
15:50:30 stacks:
15:50:30   - name: garison-test
15:50:30     class_path: blueprints.SecurityGroup.create
15:50:30     variables:
15:50:30       VpcId: ${corevpc}
15:50:30       SecurityGroupNameTag: garison-test
15:50:30       SecurityGroupDesc: garison-test Security Group
15:50:30 
15:50:30   - name: garison-test-adding-single-new-rule-to-SG
15:50:30     class_path: blueprints.SecurityGroup.addIngressRules
15:50:30     variables:
15:50:30       rules:
15:50:30         - GroupId=${output garison-test::GroupID},IpProtocol=tcp,FromPort=8080,ToPort=8080,CidrIp=${ipblock_private1}
15:50:30         - GroupId=${output garison-test::GroupID},IpProtocol=tcp,FromPort=8080,ToPort=8080,CidrIp=${ipblock_private2}
15:50:32 --------------------------------------------------------------------------------
15:50:32 Assuming Rights in AWS account ...
15:50:32 --------------------------------------------------------------------------------
15:50:33 Note: resources requiring replacement will fail this jenkins job. To allow resource replacement, check Authorize_exisiting_resource_replacement checkbox ...
15:50:34 --------------------------------------------------------------------------------
15:50:34 running stacker build using assumed rights...
15:50:34 --------------------------------------------------------------------------------
15:50:34 running: stacker build cgm.env z_garison_test-core.yaml -r us-east-1 --interactive --replacements-only -e namespace=cgm-core -t  
15:50:34 [2017-08-17T15:50:34] Using Interactive AWS Provider
15:50:34 [2017-08-17T15:50:34] Plan Status:
15:50:34 [2017-08-17T15:50:34] 	cgm-core-garison-test:                              pending
15:50:34 [2017-08-17T15:50:34] 	cgm-core-garison-test-adding-single-new-rule-to-SG: pending
15:50:34 [2017-08-17T15:50:34] Tailing stack: cgm-core-garison-test
15:50:35 [2017-08-17T15:50:35] Tailing stack: cgm-core-garison-test-adding-single-new-rule-to-SG
15:50:40 Traceback (most recent call last):
15:50:40   File "/app/jenkins/workspace/OPS_Provision-hosts-stacker-c3-equivalent_Build/venv/bin/stacker", line 9, in <module>
15:50:40     args.run(args)
15:50:40   File "/app/jenkins/workspace/OPS_Provision-hosts-stacker-c3-equivalent_Build/venv/lib/python2.7/site-packages/stacker/commands/stacker/build.py", line 52, in run
15:50:40     dump=options.dump)
15:50:40   File "/app/jenkins/workspace/OPS_Provision-hosts-stacker-c3-equivalent_Build/venv/lib/python2.7/site-packages/stacker/actions/base.py", line 127, in execute
15:50:40     self.run(*args, **kwargs)
15:50:40   File "/app/jenkins/workspace/OPS_Provision-hosts-stacker-c3-equivalent_Build/venv/lib/python2.7/site-packages/stacker/actions/build.py", line 304, in run
15:50:40     plan.execute()
15:50:40   File "/app/jenkins/workspace/OPS_Provision-hosts-stacker-c3-equivalent_Build/venv/lib/python2.7/site-packages/stacker/plan.py", line 267, in execute
15:50:40     if not self._single_run():
15:50:40   File "/app/jenkins/workspace/OPS_Provision-hosts-stacker-c3-equivalent_Build/venv/lib/python2.7/site-packages/stacker/plan.py", line 227, in _single_run
15:50:40     status = step.run()
15:50:40   File "/app/jenkins/workspace/OPS_Provision-hosts-stacker-c3-equivalent_Build/venv/lib/python2.7/site-packages/stacker/plan.py", line 74, in run
15:50:40     return self._run_func(self.stack, status=self.status)
15:50:40   File "/app/jenkins/workspace/OPS_Provision-hosts-stacker-c3-equivalent_Build/venv/lib/python2.7/site-packages/stacker/actions/build.py", line 252, in _launch_stack
15:50:40     existing_params, parameters, tags)
15:50:40   File "/app/jenkins/workspace/OPS_Provision-hosts-stacker-c3-equivalent_Build/venv/lib/python2.7/site-packages/stacker/providers/aws/interactive.py", line 270, in update_stack
15:50:40     changes = requires_replacement(changes)
15:50:40   File "/app/jenkins/workspace/OPS_Provision-hosts-stacker-c3-equivalent_Build/venv/lib/python2.7/site-packages/stacker/providers/aws/interactive.py", line 42, in requires_replacement
15:50:40     return [r for r in changeset if r["ResourceChange"]["Replacement"] ==
15:50:40 KeyError: 'Replacement'
15:50:40 Build step 'Execute shell' marked build as failure
```

The above blueprints.SecurityGroup.addIngressRules blueprint I whipped up simply adds rules to an existing security group... just a simple loop that names them "Rule1", "Rule2" etc. This was as simple as I could get my test case to report this issue.  Upon quick investigation, it appears as if this type of change should not have called the requires_replacement function... but I will look into why, later when I have a little more time.

The result of the run above DID create a changeset that was easily executable in the console. so, it was an easy work-around. (not that I really needed my test case to work, but you get my point.)

**Comment by GarisonLotus** [comment]
Some more detail here in case someone was wondering... this issue is actually happening because a changeset is generated, but no changes require replacement WHILE the --replacments-only flag is set. I suspect this is not a common scenario, which is why it went unnoticed. Fixing by checking with an `if` statement to see if the key is there at all, before checking what the key is.

**Comment by GarisonLotus** [comment]
ok, I wrote a functional test for this in PR #464, it tests by using the Dummy mock blueprint to create a simple waithandler, then runs a build again in interactive mode with `--replacements-only` set to force the function to be utilized using the Dummy2 blueprint against the same stack where a second waithandler is added and a changeset being generated showing the additional resource only.  This specific scenario failed previously as noticed above in the issue.

**Comment by GarisonLotus** [comment]
Closed issue - merged PR #464
